In [81]:
import regex as re
from collections import Counter

## 1. Замените в тексте знаки дюйма на парные кавычки-елочки. Кавычки в тексте всегда парные, вложенных кавычек нет. 

In [146]:
s = input()

 Алексей Агранович — режиссер, актер театра и кино.   Он, в частности, сыграл главную роль в фильме "Юморист" (2019 год)   режиссера Михаила Идова. В "Гоголь-центре" он   сыграл две роли в "Маленьких трагедиях", за которые   был номинирован на театральную премию "Золотая маска" за лучшую мужскую роль.


In [147]:
print(s)

Алексей Агранович — режиссер, актер театра и кино.   Он, в частности, сыграл главную роль в фильме "Юморист" (2019 год)   режиссера Михаила Идова. В "Гоголь-центре" он   сыграл две роли в "Маленьких трагедиях", за которые   был номинирован на театральную премию "Золотая маска" за лучшую мужскую роль.


In [83]:
rx = re.compile(r'"[^"]*"(*SKIP)(*FAIL)|\s')

In [84]:
new_ss = []
for word in rx.split(s):
    new_word = word.replace('"', '«',1).replace('"', '»',2)
    new_ss.append(new_word)
    new_sss = ' '.join(new_ss)
print(new_sss)

Алексей Агранович — режиссер, актер театра и кино.   Он, в частности, сыграл главную роль в фильме «Юморист» (2019 год)   режиссера Михаила Идова. В «Гоголь-центре» он   сыграл две роли в «Маленьких трагедиях», за которые   был номинирован на театральную премию «Золотая маска» за лучшую мужскую роль.


## 2. Для целого числа, если в нем больше трех разрядов, используйте пробел, чтобы сгруппировать разряды по 3.

In [148]:
a = int(input())
print(a)

 32746328477237481212


32746328477237481212


In [149]:
aa = '{:_}'.format(a).replace('_', ' ')
print(aa)

32 746 328 477 237 481 212


## 3. Повторите строку столько раз, сколько в ней символов. 

In [150]:
s = input().strip()

 Привет, мир!


In [151]:
print(s)

Привет, мир!


In [152]:
for i in range(len(s)):
    print(s)

Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!
Привет, мир!


## 4. Напишите программу, которая преобразует строку в заголовок. Части речи не учитывать: предлоги тоже преобразуются по этому правилу. 
###### Строка: Популярная блогерша и мать троих детей раскрыла секрет воспитания

In [153]:
s = input().strip().title()
print(s)

 Популярная блогерша и мать троих детей раскрыла секрет воспитания


Популярная Блогерша И Мать Троих Детей Раскрыла Секрет Воспитания


## 5. На вход подается текст, состоящий из строк (символ-разделитель: \n). Пронумеруйте строки, отдав под номер 4 символа. В качестве заполнителя используйте пробелы. 

И скучно и грустно, и некому руку подать  
В минуту душевной невзгоды…  
Желанья!.. что пользы напрасно и вечно желать?..  
А годы проходят — все лучшие годы!  
 
Любить… но кого же?.. на время — не стоит труда,  
А вечно любить невозможно.  
В себя ли заглянешь? — там прошлого нет и следа:  
И радость, и муки, и всё там ничтожно…  
 
Что страсти? — ведь рано иль поздно их сладкий недуг  
Исчезнет при слове рассудка;  
И жизнь, как посмотришь с холодным вниманьем вокруг —  
Такая пустая и глупая шутка…

In [154]:
s = input().strip()

 И скучно и грустно, и некому руку подать   В минуту душевной невзгоды…   Желанья!.. что пользы напрасно и вечно желать?..   А годы проходят — все лучшие годы!     Любить… но кого же?.. на время — не стоит труда,   А вечно любить невозможно.   В себя ли заглянешь? — там прошлого нет и следа:   И радость, и муки, и всё там ничтожно…     Что страсти? — ведь рано иль поздно их сладкий недуг   Исчезнет при слове рассудка;   И жизнь, как посмотришь с холодным вниманьем вокруг —   Такая пустая и глупая шутка…


In [155]:
ss = re.split(r'\s\s', s)

In [156]:
ss

['И скучно и грустно, и некому руку подать',
 ' В минуту душевной невзгоды…',
 ' Желанья!.. что пользы напрасно и вечно желать?..',
 ' А годы проходят — все лучшие годы!',
 '',
 ' Любить… но кого же?.. на время — не стоит труда,',
 ' А вечно любить невозможно.',
 ' В себя ли заглянешь? — там прошлого нет и следа:',
 ' И радость, и муки, и всё там ничтожно…',
 '',
 ' Что страсти? — ведь рано иль поздно их сладкий недуг',
 ' Исчезнет при слове рассудка;',
 ' И жизнь, как посмотришь с холодным вниманьем вокруг —',
 ' Такая пустая и глупая шутка…']

In [157]:
for i, val in enumerate(ss, start=1):
    print(i, val)

1 И скучно и грустно, и некому руку подать
2  В минуту душевной невзгоды…
3  Желанья!.. что пользы напрасно и вечно желать?..
4  А годы проходят — все лучшие годы!
5 
6  Любить… но кого же?.. на время — не стоит труда,
7  А вечно любить невозможно.
8  В себя ли заглянешь? — там прошлого нет и следа:
9  И радость, и муки, и всё там ничтожно…
10 
11  Что страсти? — ведь рано иль поздно их сладкий недуг
12  Исчезнет при слове рассудка;
13  И жизнь, как посмотришь с холодным вниманьем вокруг —
14  Такая пустая и глупая шутка…


## 6. Дан чек, в котором перечислены купленные товары. В каждой второй строке — целочисленная стоимость товара. Вычислите полную сумму покупки. 
Молоко МФ ТАЛИЦКИЕ паст Дерев 3,5-4%1000  
82  
Пакет ЛЕНТА майка 12кг  
6  
Хлеб СМАК Таежный ржаной 350 г  
27  
Сельдь МЕРИДИАН Дингли ф/к в бел вине115  
99  
Краб пал МИРАМАР Кл им кр мяс охл 200гр  
72  
Батон СМАК Подмосковный в/с нар 380г  
33  
Макароны BARILLA Spaghetti n.5 450г  
86  
Макароны BORGES Spagetti 500г  
159  
Салат Оливье классический лоток вес  
81  
Салат Свекла с чесноком лоток вес  
85  
Колбаса МЯСНАЯ КЛАССИКА Серв Свердл 400  
211

In [158]:
chek = input()

 Молоко МФ ТАЛИЦКИЕ паст Дерев 3,5-4%1000   82   Пакет ЛЕНТА майка 12кг   6   Хлеб СМАК Таежный ржаной 350 г   27   Сельдь МЕРИДИАН Дингли ф/к в бел вине115   99   Краб пал МИРАМАР Кл им кр мяс охл 200гр   72   Батон СМАК Подмосковный в/с нар 380г   33   Макароны BARILLA Spaghetti n.5 450г   86   Макароны BORGES Spagetti 500г   159   Салат Оливье классический лоток вес   81   Салат Свекла с чесноком лоток вес   85   Колбаса МЯСНАЯ КЛАССИКА Серв Свердл 400   211


In [159]:
chek_split = re.split(r'\s\s', chek)

In [160]:
list_chek = []
for i in chek_split:
    if i.strip().isdigit():
        list_chek.append(int(i))
print('Сумма покупок: {}'.format(sum(list_chek)))

Сумма покупок: 941


## 7. Напишите программу, которая «оборачивает» телефон в теги

In [161]:
tel = input()
print(tel)

 +79012342452


+79012342452


In [162]:
#1 вариант
href = '<a href="tel:%s">%s</a>' % (tel, tel)
href

'<a href="tel:+79012342452">+79012342452</a>'

In [163]:
#2 вариант
print(f'<a href="tel:{tel}">{tel}</a>')

<a href="tel:+79012342452">+79012342452</a>


## 8. Дан секвенированный геном какого-либо вируса. 

###### Нужно определить количество нуклеотидов a c g t в геноме в таком порядке. 
        1 accggatggc cgcgattttt tttcccgtac acgctccaaa gtaaaagtaa agtaactagt  
       61 ggtccccgcc cactaacaaa tgtcagccaa tcagaatgaa gcctgaaagc tttcttatct  
      121 agattttgtc tttatatact tgctcgccaa gtagttttat tgcaacatgt gggacacatt  
      181 attgaacgag tttccagact ctgttcatgg ttttcgttgt atgctgggta taaaatatct  
      241 gcagctttta gaagaggagt acgagcccaa tactttggga cacgatttaa ttagagatct  
      301 catatctgtc attagggcta agaattatgt cgaagcgacc cggcgatata ataattttct  
      361 gaggaaaatc ctagtgtttt ccgacgagtg aaggcaaaga ttgcagagga agaattcctc  
      421 ctgaaattca aatttcaaat ttgaaatctt ggcaattgcg cctcaaacag ttgctgtcaa  
      481 gggacccaga tgaccgcact atcttttggg tgtatggacc caccggagga gagggcaaat  
      541 cgacatttgc ccgtgacctg tacagaagtg ggtcctggtt ctatacacgt ggaggatctg  
      601 ctgataatgt ggcttaccag tatataggat gtttaggaaa taatatagtc tttgatatcc  
      661 ctagggataa aaaagattat ttacaatata gtttagtaga aatgtttaag gatagattag  
      721 tagttagtaa taagtatgaa ccgataatgg ctcctatgct taattgtatt catgtagtag  
      781 ttatgtctaa ttttcttcca gatatggaga agattagtag tgatagggtc cgtgtaattc  
      841 cttgtattcc ttgtggtgtt tgtcttaagc atcatgatca taatataaaa tgtgatgagt  
      901 atttagaata aataaaatga ctgttgaatt tgatattatt taagaaaaca aggccccgca  
      961 ggggcaatat aaaaaaatat tttgcttctt gaaaacaaga aggaatgata tgaaagaaat  
     1021 ataagtgggt ctcgaattgc ataagcacat ggagatgagg ttcgccattc tcgtggagtt  
     1081 ctctgcaaat tttaatatat tttttattgg ttggggtatt taatttgagt atttgttcga  
     1141 gtgcctcttc tttggataga gagcattttg ggaaagtgag gaaataattt tttgcattta  
     1201 taagaaattt tcgttgggta ggcatgttgg tcagaggacc cgattgactc tctattcaat  
     1261 ctttcatatg tatcgggtcc ctgggtcctt atttatatgt gaggacctta atggcattat  
     1321 tgtaattagg taaaagtaat acgctttaat tcaaattggt aaagcggcca tccgtataat  
     1381 att

In [164]:
s = input()

         1 accggatggc cgcgattttt tttcccgtac acgctccaaa gtaaaagtaa agtaactagt          61 ggtccccgcc cactaacaaa tgtcagccaa tcagaatgaa gcctgaaagc tttcttatct         121 agattttgtc tttatatact tgctcgccaa gtagttttat tgcaacatgt gggacacatt         181 attgaacgag tttccagact ctgttcatgg ttttcgttgt atgctgggta taaaatatct         241 gcagctttta gaagaggagt acgagcccaa tactttggga cacgatttaa ttagagatct         301 catatctgtc attagggcta agaattatgt cgaagcgacc cggcgatata ataattttct         361 gaggaaaatc ctagtgtttt ccgacgagtg aaggcaaaga ttgcagagga agaattcctc         421 ctgaaattca aatttcaaat ttgaaatctt ggcaattgcg cctcaaacag ttgctgtcaa         481 gggacccaga tgaccgcact atcttttggg tgtatggacc caccggagga gagggcaaat         541 cgacatttgc ccgtgacctg tacagaagtg ggtcctggtt ctatacacgt ggaggatctg         601 ctgataatgt ggcttaccag tatataggat gtttaggaaa taatatagtc tttgatatcc         661 ctagggataa aaaagattat ttacaatata gtttagtaga aatgtttaag gatagattag         721 tagttagtaa taagtatgaa ccgataatgg ctcctatgct taattgtat

In [165]:
ss = [i for i in s if i.isalpha()]

In [166]:
a = [val for val in Counter(ss).values()]
print(f'Количество нуклеотидов a c g t в геноме: {a}')

Количество нуклеотидов a c g t в геноме: [415, 212, 303, 453]


## 9. Дан текст с меню, 
###### в котором указана цена в долларах (всегда целое число). Перевести его по курсу $1 = 75р в рубли. Воспользуйтесь предыдущим кодом: сумму в рублях делите по группам из 3 разрядов.

###### В выводе буква р - это кириллица. 

In [167]:
s = input()
print(s)

 Bigoli con ragoutd’anatra, fungo shiitake e salvia frittaBigoli   with duck ragout, shiitake mushrooms and fried sage $18     Spaghetticon pesto di topinambur e pistacchio, limone esale al timo   Spaghetti with topiambur and pistachios pesto sauce $17


Bigoli con ragoutd’anatra, fungo shiitake e salvia frittaBigoli   with duck ragout, shiitake mushrooms and fried sage $18     Spaghetticon pesto di topinambur e pistacchio, limone esale al timo   Spaghetti with topiambur and pistachios pesto sauce $17


In [168]:
ss = re.split(r'\s', s)

In [169]:
print(ss)

['Bigoli', 'con', 'ragoutd’anatra,', 'fungo', 'shiitake', 'e', 'salvia', 'frittaBigoli', '', '', 'with', 'duck', 'ragout,', 'shiitake', 'mushrooms', 'and', 'fried', 'sage', '$18', '', '', '', '', 'Spaghetticon', 'pesto', 'di', 'topinambur', 'e', 'pistacchio,', 'limone', 'esale', 'al', 'timo', '', '', 'Spaghetti', 'with', 'topiambur', 'and', 'pistachios', 'pesto', 'sauce', '$17']


In [170]:
#change prices in dollars to rubles
price_new = []
for string in ss:
    if string.startswith('$'):
        string_ = string.replace('$', '')
        number = int(string_)*75
        number_ = '{:_}р'.format(number).replace('_', ' ')
        price_new.append(number_)
    else:
        price_new.append(string)
price_new_new = ' '.join(price_new)

In [171]:
print(price_new_new)

Bigoli con ragoutd’anatra, fungo shiitake e salvia frittaBigoli   with duck ragout, shiitake mushrooms and fried sage 1 350р     Spaghetticon pesto di topinambur e pistacchio, limone esale al timo   Spaghetti with topiambur and pistachios pesto sauce 1 275р


## 10. Напишите программу, которая для заданного штрих-кода говорит, правильный он (yes) или нет (no)
###### Например, вот мой кетчуп. Код у него — 4604075024433. На нечетных местах: 4, 0, 0, 5, 2, 4, 3. На четных: 6, 4, 7, 0, 4, 3. Если мы сложим первую сумму с утроенной второй получится 90. Значит, это корректный штрих-код. 

In [172]:
w = input().strip()
print(w)

 4604075024433


4604075024433


In [173]:
#find eval and uneval numbers' indexes
w_eval = [i for i in w[1::2]]
w_uneval = [i for i in w[0::2]]

In [174]:
print(w_uneval)
print(w_eval)

['4', '0', '0', '5', '2', '4', '3']
['6', '4', '7', '0', '4', '3']


In [177]:
#make sum of all eval and uneval numbers' indexes
res = sum(int(i) for i in w_uneval) + \
(sum(int(i) for i in w_eval))*3
print(res)

90


In [178]:
#chek the sum
if res % 10 == 0:
    print('yes')
else:
    print('no')

yes


## 11. Напишите в порядке возрастания все делители числа, кроме 1 и его самого. Возможно, у вас будет ошибка лимита по времени, попытайтесь немного оптимизировать программу. 

In [185]:
a = int(input())
print(a)

 500


500


In [186]:
aa = [i for i in range(2,a) if a%i == 0]

In [187]:
print(aa)

[2, 4, 5, 10, 20, 25, 50, 100, 125, 250]


## 12. Напишите программу, которая умеет склонять с любыми числительными слово «комментарий». 

In [188]:
d = input()

 1345981346


In [189]:
#make lists with appropriate endings to nouns
ev = list(str(i) for i in range(5,20))+\
list('0')
ya = list(str(i) for i in range(2,5))

In [190]:
print(ev)
print(ya)

['5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '0']
['2', '3', '4']


In [191]:
if d.endswith(tuple(ev)):
    print(f'{d} комментариев')
elif d.endswith(tuple(ya)):
    print(f'{d} комментария')
else:
    print(f'{d} комментарий')

1345981346 комментариев
